cromshell

```diff
@@ -731,18 +735,42 @@ function status()
   r=$?
   [[ $r -eq 0 ]] && retVal=1
 
+  # Hold our status string here:
+  local workflowStatus=$( cat $f | jq . | grep status | sed -e 's#.*: ##g' | tr -d '",' )
+
   if [[ $retVal -eq 1 ]]; then
     turtleDead
+  elif [[ "${workflowStatus}" == "Running" ]] ; then
+    # OK, status claims this workflow is running fine, but we need to check to see
+    # if there are any failed sub-processes.
+    # To do this, we use the `execution-status-count` logic with some filtering:
+    local tmpExecutionStatusCount=$( makeTemp )
+    local tmpMetadata=$( makeTemp )
+
+    # Get execution status count and filter the metadata down:
+    curl --connect-timeout "$CURL_CONNECT_TIMEOUT" --max-time "$CURL_MAX_TIMEOUT" --compressed -s "${2}/api/workflows/v1/${1}/metadata?$CROMWELL_SLIM_METADATA_PARAMETERS" | jq '.calls | map_values(group_by(.executionStatus) | map({(.[0].executionStatus): . | length}) | add)' > ${tmpExecutionStatusCount}
```

Comment:
I think we drop some greps and a few lines by replacing this block with something like:
```
curl ... | jq --exit-status '[ ..|.executionStatus? | values | . == "Fail" ] | any' >  /dev/null
```

That will exit with 0 if any value is `Fail` and 1 if no value is fail.  We can invert it with `. !+ "Fail" | all`

Reply:
We can make the check more elegant, but I'm caching the execution-status-count results so I can display them to the user later.

I'll still need to keep that file / call around.

Reply:
Fixed.